src/rules/unnecessaryElseRule.ts

```diff
@@ -21,15 +21,26 @@ import * as Lint from "../index";
 
 import { codeExamples } from "./code-examples/unnecessaryElse.examples";
 
+const OPTION_ALLOW_ELSE_IF = "allow-else-if";
+
 export class Rule extends Lint.Rules.AbstractRule {
     /* tslint:disable:object-literal-sort-keys */
     public static metadata: Lint.IRuleMetadata = {
         description: Lint.Utils.dedent`
         Disallows \`else\` blocks following \`if\` blocks ending with a \`break\`, \`continue\`, \`return\`, or \`throw\` statement.`,
         descriptionDetails: "",
-        optionExamples: [true],
-        options: null,
-        optionsDescription: "Not configurable.",
+        optionExamples: [true, [true, OPTION_ALLOW_ELSE_IF]],
+        options: {
+            type: "array",
+            items: {
+                type: "string",
+                enum: [OPTION_ALLOW_ELSE_IF],
+            },
+            uniqueItems: true,
+        },
+        optionsDescription: Lint.Utils.dedent`
+            You can optionally specify the option \`"${OPTION_ALLOW_ELSE_IF}"\` to allow "else if" statements.
```

Comment:
TSLint's been moving towards having options as objects with members, rather than single primitives. They're a lot easier to add more options to later on and a bit more clear in `tslint.json`s. You'll want to switch this to `type: "object"`; see `memberOrderingRule.ts` and `noShawdowedVariableRule.ts` as examples.

Reply:
Cool, thanks for pointing an examples.

Reply:
Please take a look at https://github.com/palantir/tslint/pull/4599/commits/88397d98c549b7016704b6e44dbf3817bec6f648 - I'm not sure that I've done it right (especially in `optionsDescription` prop).

Reply:
Yup, looks good!